zcnbridge/authorizers_query.go

```diff
@@ -0,0 +1,213 @@
+package zcnbridge
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"net/http"
+	u "net/url"
+	"strings"
+	"sync"
+
+	"github.com/0chain/gosdk/zcnbridge/zcnsc"
+
+	"github.com/0chain/gosdk/zcnbridge/log"
+	"go.uber.org/zap"
+
+	"github.com/0chain/gosdk/zcnbridge/errors"
+	bridge "github.com/0chain/gosdk/zcnbridge/http"
+	"github.com/0chain/gosdk/zcnbridge/node"
+	"github.com/0chain/gosdk/zcnbridge/wallet"
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+type (
+	// authorizerResponse is HTTP client response event
+	authorizerResponse struct {
+		// 	AuthorizerID is authorizer where the job was performed
+		AuthorizerID string
+		// event is server job event
+		event JobResult
+		// error describes an error occurred during event processing on client side
+		error
+	}
+
+	requestHandler struct {
+		path    string
+		values  u.Values
+		decoder func([]byte) (JobResult, error)
+	}
+
+	responseChannelType chan *authorizerResponse
+	eventsChannelType   chan []JobResult
+)
+
+var (
+	client *retryablehttp.Client
+)
+
+// CreateZCNMintPayload gets burn ticket and creates mint payload to be minted in the chain
+func CreateZCNMintPayload(hash string) (*zcnsc.MintPayload, error) {
+	client = bridge.NewRetryableClient()
+	authorizers, err := GetAuthorizers()
+
+	if err != nil || len(authorizers.NodeMap) == 0 {
+		return nil, errors.Wrap("get_authorizers", "failed to get authorizers", err)
+	}
+
+	var (
+		totalWorkers = len(authorizers.NodeMap)
+		values       = u.Values{
+			"hash":     []string{hash},
+			"address":  []string{wallet.ZCNSCSmartContractAddress},
+			"clientid": []string{node.ID()},
+		}
+	)
+
+	handler := &requestHandler{
+		path:   wallet.BurnWzcnTicketPath,
+		values: values,
+		decoder: func(body []byte) (JobResult, error) {
+			ev := &WZCNBurnEvent{}
+			err := json.Unmarshal(body, ev)
+			return ev, err
+		},
+	}
+
+	results := queryAllAuthorizers(authorizers, handler)
+
+	numSuccess := len(results)
+
+	quorum := math.Ceil((float64(numSuccess) * 100) / float64(totalWorkers))
+
+	if numSuccess > 0 && quorum >= wallet.ConsensusThresh && len(results) > 1 {
+		burnTicket, ok := results[0].Data().(*proofEthereumBurn)
+		if !ok {
+			return nil, errors.Wrap("type_cast", "failed to convert to *proofEthereumBurn", err)
+		}
+
+		var sigs []*zcnsc.AuthorizerSignature
```

Comment:
As I understood Authorizers should authorize burn tickets, how long it will take if the client calls it? And how many Authorizers required?

Reply:
It depends on the transaction confirmed. If Ethereum transaction is confirmed then this will be fast 1-10 seconds. For ethereum transaction this is fast. We will have for start 20 authorizers. All authorizer requests will be run async concurrently @moldis

Reply:
@Dmdv Even 10 seconds is slow, and I guess it might take more time and not depend on us, more on eth network. Any way we can move this to the backend side and let client just sign the transaction, then give back some sort of hash to check transactions status?